code/controllers/subsystem/blackbox.dm

```diff
@@ -0,0 +1,307 @@
+  *
+  * Called during [/proc/broadcast_message()] to log a message to the blackbox.
+  * Translates the specific frequency to a name
+  *
+  * Arguments:
+  * * freq - Frequency of the transmission
+  */
+/datum/controller/subsystem/blackbox/proc/LogBroadcast(freq)
+	if(sealed)
+		return
+	switch(freq)
+		if(PUB_FREQ)
+			record_feedback("tally", "radio_usage", 1, "common")
+		if(SCI_FREQ)
+			record_feedback("tally", "radio_usage", 1, "science")
+		if(COMM_FREQ)
+			record_feedback("tally", "radio_usage", 1, "command")
+		if(MED_FREQ)
+			record_feedback("tally", "radio_usage", 1, "medical")
+		if(ENG_FREQ)
+			record_feedback("tally", "radio_usage", 1, "engineering")
+		if(SEC_FREQ)
+			record_feedback("tally", "radio_usage", 1, "security")
+		if(DTH_FREQ)
+			record_feedback("tally", "radio_usage", 1, "deathsquad")
+		if(SYND_FREQ)
+			record_feedback("tally", "radio_usage", 1, "syndicate")
+		if(SYNDTEAM_FREQ)
+			record_feedback("tally", "radio_usage", 1, "syndicate team")
+		if(SUP_FREQ)
+			record_feedback("tally", "radio_usage", 1, "supply")
+		if(SRV_FREQ)
+			record_feedback("tally", "radio_usage", 1, "service")
+		if(PROC_FREQ)
+			record_feedback("tally", "radio_usage", 1, "procedure")
+		else
+			record_feedback("tally", "radio_usage", 1, "other")
+
+
+/**
+  * Helper to find and return a feeedback datum
+  *
+  * Pass in a feedback datum key and key_type to do a lookup.
+  * It will create the feedback datum if it doesnt exist
+  *
+  * Arguments:
+  * * key - Key of the variable to lookup
+  * * key_type - Type of feedback to be recorded if the feedback datum cant be found
+  */
+/datum/controller/subsystem/blackbox/proc/find_feedback_datum(key, key_type)
+	for(var/datum/feedback_variable/FV in feedback)
+		if(FV.key == key)
+			return FV
+
+	var/datum/feedback_variable/FV = new(key, key_type)
+	feedback += FV
+	return FV
+
+/**
+  * Main feedback recording proc
+  *
+  * This is the bulk of this subsystem and is in charge of creating and using the variables.
+  * See .github/USING_FEEDBACK_DATA.md for instructions
+  * Note that feedback is not recorded to the DB during this function. That happens at round end.
+  *
+  * Arguments:
+  * * key_type - Type of key. Either "text", "amount", "tally", "nested tally", "associative"
+  * * key - Key of the data to be used (EG: "admin_verb")
+  * * increment - If using "amount", how much to increment why
+  * * data - The actual data to logged
+  * * overwrite - Do we want to overwrite the existing key
+  */
+/datum/controller/subsystem/blackbox/proc/record_feedback(key_type, key, increment, data, overwrite)
+	if(sealed || !key_type || !istext(key) || !isnum(increment || !data))
+		return
+	var/datum/feedback_variable/FV = find_feedback_datum(key, key_type)
+	switch(key_type)
+		if("text")
+			if(!istext(data))
+				return
+			if(!islist(FV.json["data"]))
+				FV.json["data"] = list()
+			if(overwrite)
+				FV.json["data"] = data
+			else
+				FV.json["data"] |= data
```

Comment:
Do we maybe want
```suggestion
			if(overwrite)
				FV.json["data"] = "[data]"
			else
				FV.json["data"] |= "[data]"
```
for when people inevitably accidentally a wrong type?

Reply:
Types will be auto encoded
![image](https://user-images.githubusercontent.com/25063394/103083961-a72c9700-45d5-11eb-8904-605f15f2c3b1.png)